app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  def create
+    commentable = set_commentable
+    comment = commentable.comments.build(comment_params)
+    comment.user = current_user
+
+    if comment.save
+      redirect_to commentable, notice: 'コメントを作成しました'
+    else
+      redirect_to commentable, alert: 'コメントの作成に失敗しました'
+    end
+  end
+
+  def destroy
+    @comment = current_user.comments.find_by(id: params[:comment_id])
+    redirect_to @comment.commentable, alert: '他の人のコメントは削除できません' if @comment.nil?
```

Comment:
```suggestion
    @comment = current_user.comments.find(params[:comment_id])
```

通常の操作はありえないエラーなので、404エラーを返す形で良いと思います。